java_classes_block/main_task/src/main/java/Main.java

```diff
@@ -0,0 +1,97 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.List;
+
+public class Main { //Main tasks in Java Classes block
+
+    public static void main(String[] args) {
+
+//        6. House: id, Номер квартиры, Площадь, Этаж, Количество комнат, Улица, Тип здания, Срок эксплуатации.
+//         Создать массив объектов. Вывести:
+
+        House house1 = new House.HouseBuilder(1)
+                .setFlatNumber(343)
+                .setFloor(15)
+                .setSquare(79.8)
+                .setStreet("Lenina")
+                .setRoomNumber(3)
+                .build();
+
+        House house2 = new House.HouseBuilder(2)
+                .setFlatNumber(79)
+                .setFloor(7)
+                .setSquare(56.9)
+                .setStreet("Lenina")
+                .setRoomNumber(2)
+                .build();
+
+        House house3 = new House.HouseBuilder(3)
+                .setFlatNumber(2)
+                .setFloor(5)
+                .setSquare(25.7)
+                .setStreet("Lenina")
+                .setRoomNumber(1)
+                .build();
+
+        House house4 = new House.HouseBuilder(4)
+                .setFlatNumber(57)
+                .setFloor(3)
+                .setSquare(35.4)
+                .setRoomNumber(2)
+                .setStreet("Lenina")
+                .build();
+
+        House house5 = new House.HouseBuilder(5)
+                .setFlatNumber(196)
+                .setFloor(9)
+                .setSquare(20.3)
+                .setRoomNumber(1)
+                .setStreet("Lenina")
+                .build();
+
+        List<House> initialHouseList = Arrays.asList(house1, house2, house3, house4, house5);
+
+        // a) список квартир, имеющих заданное число комнат;
+        System.out.println("Enter the number of rooms:");
+        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+        try {
+            int  numberOfRooms = Integer.parseInt(reader.readLine());
+            List<House> byNumberOfRoomsHouseList = HouseUtils.getHousesByNumberOfRooms(numberOfRooms, initialHouseList);
+            for (House house : byNumberOfRoomsHouseList) {
+                System.out.println(house.toString());
+            }
+        } catch (IOException e) {
+            throw new IllegalStateException("Cannot find houses with the entered number of rooms", e);
```

Comment:
in fact `IOException` gives only one line
```
reader.readLine()
```

and you use it in 3 places
```
            int  numberOfRooms = Integer.parseInt(reader.readLine());
            int  minSquare = Integer.parseInt(reader.readLine());
            int  numberOfRooms = Integer.parseInt(reader.readLine());
```

So I suggest extraction of this line into a separate function which will contain try-catch inside so you will get rid of 3 try-catch blocks

Reply:
Did method readIntFromConsole in HouseUtils class